lib/value-pairs/value-pairs.c

```diff
@@ -617,20 +617,26 @@ vp_walker_split_name_to_tokens(vp_walk_state_t *state, const gchar *name)
 
   GPtrArray *array = g_ptr_array_sized_new(VP_STACK_INITIAL_SIZE);
 
-  do
+  while (pos < name_len)
     {
-      pos += strcspn(&name[pos], "@.");
       if (name[pos] == '@')
-        pos = vp_walker_skip_sdata_enterprise_id(name, pos);
-
-      if (name[pos] == '.' || pos == name_len)
+        {
+          pos = vp_walker_skip_sdata_enterprise_id(name, pos);
+        }
+      else if (name[pos] == '.')
         {
           g_ptr_array_add(array, g_strndup(&name[token_start], pos - token_start));
           pos++;
           token_start = pos;
         }
+      else
+        {
+          pos += strcspn(&name[pos], "@.");
+        }
     }
-  while (pos < name_len);
+
+  if ((pos - token_start) > 0)
+    g_ptr_array_add(array, g_strndup(&name[token_start], pos - token_start));
 
```

Comment:
The other version would be a bit more complex body of the loop.
I think this one is the lesser of two evils.